### PR TITLE
fix: resolve remaining FSD feature boundaries

### PR DESCRIPTION
**Pull request by JoelA510**
## Summary
- resolves the final Wave D FSD lateral dependencies by moving template-search and related-template logic behind shared adapters
- passes project team/member data down through page and panel composition instead of importing people hooks inside task components
- keeps library feature wrappers in place for existing library callers while task/project features consume legal shared boundaries
- fixes the Build & Test workflow unit-test command so it uses the existing 
pm test run-mode script instead of invoking itest --run --run

## Validation
CI-equivalent Build & Test sequence run locally:
- 
pm ci
- AGENT_MODE=true npm run verify-architecture
- 
pm run lint
- VITE_SUPABASE_URL=https://placeholder.supabase.co VITE_SUPABASE_ANON_KEY=placeholder-key npm test
- 
pm run db:local:test
- 
pm run build

Additional checks:
- git diff --check passed with Git CRLF normalization warnings only
- ctionlint was not installed locally, so no actionlint run was performed

## Notes
- No Playwright E2E was run.
- Existing lint warnings remain unchanged.
- 
pm run db:local:test passed with Docker available; Supabase CLI printed its standard newer-version notice.

### CODE REVIEW

File: src/features/tasks/components/StrategyFollowUpDialog.tsx [hunk 1]
```diff
@@ -31,6 +38,136 @@ interface StrategyFollowUpDialogProps {
...
+        setQuery(template.title ?? '');
+        setIsOpen(false);
+        setActiveIndex(-1);
+    }, [onSelect]);
+
+    const handleKeyDown = useCallback((event: KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === 'ArrowDown') {
+            event.preventDefault();
+            setIsOpen(true);
+            setActiveIndex((prev) => (prev + 1 >= results.length ? 0 : prev + 1));
```
[the remark this answers is not in the document]
Comment (JoelA510) [reply src/features/tasks/components/StrategyFollowUpDialog.tsx]: Approved and remediated in cdd9244. The ArrowDown handler now preserves `activeIndex = -1` when there are no search results, so the combobox never points at a nonexistent option. I also added a focused unit test for the empty-results keyboard path.

File: src/features/tasks/components/StrategyFollowUpDialog.tsx [hunk 2]
```diff
@@ -31,6 +38,136 @@ interface StrategyFollowUpDialogProps {
...
+            <label
+                className="block text-sm font-medium text-slate-600"
+                htmlFor={`strategy-template-search-${listboxId}`}
+            >
+                Search Master Library
+            </label>
+            <input
+                id={`strategy-template-search-${listboxId}`}
+                type="text"
+                className="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring md:text-sm"
```
[the remark this answers is not in the document]
Comment (JoelA510) [reply src/features/tasks/components/StrategyFollowUpDialog.tsx]: Approved and remediated in fa9c5a3. The search input now uses `h-10` for the touch target and `focus-visible:ring-brand-500`, matching the cited style-guide pattern. I reran the StrategyFollowUpDialog unit test file after the change.

File: src/features/tasks/components/StrategyFollowUpDialog.tsx [hunk 3]
```diff
@@ -31,6 +38,136 @@ interface StrategyFollowUpDialogProps {
...
+                aria-expanded={isOpen && hasResults}
+                aria-haspopup="listbox"
+            />
+
+            {isOpen && (
+                <div
+                    id={listboxId}
+                    role="listbox"
+                    aria-label="Master Library search results"
+                    className="absolute z-50 mt-1 w-full max-h-64 overflow-y-auto rounded-md border border-slate-200 bg-white shadow-lg"
```
[the remark this answers is not in the document]
Comment (JoelA510) [reply src/features/tasks/components/StrategyFollowUpDialog.tsx]: Approved and remediated in fd79de9. The search result listbox now uses `rounded-xl` so the dropdown matches the app card/list surface style called out in the review. I reran the StrategyFollowUpDialog unit test file after the change.

File: src/features/tasks/components/StrategyFollowUpDialog.tsx [hunk 4]
```diff
@@ -31,6 +38,136 @@ interface StrategyFollowUpDialogProps {
...
+                            <button
+                                key={template.id}
+                                type="button"
+                                id={`${listboxId}-item-${template.id}`}
+                                role="option"
+                                aria-selected={isActive}
+                                onMouseDown={(event) => event.preventDefault()}
+                                onClick={() => handleSelect(template)}
+                                data-testid={`strategy-followup-search-row-${template.id}`}
+                                className={`w-full text-left px-4 py-2.5 border-b border-slate-50 last:border-b-0 focus:outline-none ${
```
[the remark this answers is not in the document]
Comment (JoelA510) [reply src/features/tasks/components/StrategyFollowUpDialog.tsx]: Approved and remediated in a13f5d8. The option rows now use `border-slate-200`, which keeps the separator visible and aligned with the standard border token. I reran the StrategyFollowUpDialog unit test file after the change.

File: src/features/tasks/components/StrategyFollowUpDialog.tsx [hunk 5]
```diff
@@ -31,6 +38,136 @@ interface StrategyFollowUpDialogProps {
...
+                                    isActive ? 'bg-brand-50' : 'hover:bg-slate-50'
+                                }`}
+                            >
+                                <p className="text-sm font-medium text-slate-900 truncate">{template.title}</p>
+                                {template.description && (
+                                    <p className="text-xs text-slate-500 truncate mt-0.5">{template.description}</p>
```
[the remark this answers is not in the document]
Comment (JoelA510) [reply src/features/tasks/components/StrategyFollowUpDialog.tsx]: Approved and remediated in 825c667. Template descriptions now use `text-slate-600`, matching the style-guide body text color while preserving the existing layout and truncation behavior. I reran the StrategyFollowUpDialog unit test file after the change.

File: src/features/tasks/components/StrategyFollowUpDialog.tsx [hunk 6]
```diff
@@ -31,6 +38,136 @@ interface StrategyFollowUpDialogProps {
...
+                        <div className="px-4 py-3 text-sm text-slate-500">Loading templates…</div>
+                    )}
+
+                    {!isLoading && results.length === 0 && (
+                        <div className="px-4 py-3 text-sm text-slate-500">
+                            {exclusionDrained
+                                ? 'All matching templates are already added.'
+                                : query
+                                    ? 'No matching templates found.'
+                                    : 'No templates available.'}
```
[the remark this answers is not in the document]
Comment (JoelA510) [reply src/features/tasks/components/StrategyFollowUpDialog.tsx]: Approved and remediated in 5ac735e. The inline StrategyTemplateSearch copy now goes through `react-i18next` keys, reuses the existing task search placeholder key, and adds matching en/es entries for the follow-up search label, result status messages, aria label, and action copy. I also ran the dialog test and the en/es key-parity test for this change.
